crypto/fipsmodule/rand/entropy/tree_drbg_jitter_entropy.c

```diff
@@ -0,0 +1,466 @@
+// entropy source for the frontend DRBGs. The implemented tree-DRBG has the
+// following characteristics:
+//  - A per-thread seed DRBG that serves seed requests for a thread-local
+//    frontend DRBG.
+//  - A global seed DRBG that serves seed requests from the thread-local seed
+//    DRBGs.
+//  - A root seed source that serves seed requests from the global seed DRBG.
+//    The root seed source is a global instance of Jitter Entropy.
+//
+// The dependency tree looks as follows:
+//
+//          entropy_source
+//            interface
+//                |
+//   new_rand.c   |   tree_drbg_jitter_entropy.c
+//                |
+//  front-end     |   tree-DRBG
+//  per-thread    |   per-thread
+// +-----------+  |  +-----------+
+// | CTR-DRBG  | --> | CTR-DRBG  | -|
+// +-----------+  |  +-----------+   -|
+// +-----------+  |  +-----------+     --|     per-process         per-process
+// | CTR-DRBG  | --> | CTR-DRBG  | ---|   --> +-----------+     +---------------+
+// +-----------+  |  +-----------+     -----> | CTR-DRBG  | --> |Jitter Entropy |
+//      ...       |      ...              --> +-----------+     +---------------+
+// +-----------+  |  +-----------+  -----|
+// | CTR-DRBG  | --> | CTR-DRBG  |-|
+// +-----------+  |  +-----------+
+//                |
+//
+// Memory life-cycle: The thread-local DRBGs has the same storage duration as
+// their corresponding thread-local frontend DRBGs. The per-process DRBG and
+// Jitter Entropy instance has a storage duration that extends to the duration
+// of AWS-LC being loaded into the process. The per-process memory is lazily
+// allocated.
+
+// To serve seed requests from the frontend DRBGs the following
+// |struct entropy_source_methods| interface functions are implemented:
+//  - tree_jitter_initialize
+//  - tree_jitter_zeroize_thread_drbg
+//  - tree_jitter_free_thread_drbg
+//  - tree_jitter_get_seed
+
+
+// Placeholder right now until SP800-90C is ratified that might specify these
+// further.
+// TREE_JITTER_GLOBAL_DRBG_MAX_GENERATE = 2^24
+#define TREE_JITTER_GLOBAL_DRBG_MAX_GENERATE 0xFFFFFF
+// TREE_JITTER_THREAD_DRBG_MAX_GENERATE = 2^20
+#define TREE_JITTER_THREAD_DRBG_MAX_GENERATE 0xFFFFF
+
+struct tree_jitter_drbg_t {
+  // is_global is 1 if this object is the per-process seed DRBG. Otherwise 0.
+  uint8_t is_global;
+
+  // drbg is the DRBG state.
+  CTR_DRBG_STATE drbg;
+
+  // max_generate_calls is the maximum number of generate calls that can be
+  // invoked on |drbg| without a reseed.
+  uint64_t max_generate_calls;
+
+  // reseed_calls_since_initialization is the number of reseed calls made of
+  // |drbg| since its initialization.
+  uint64_t reseed_calls_since_initialization;
+
+  // generation_number caches the UBE generation number.
+  uint64_t generation_number;
+
+  // ube_protection denotes whether this object is protected from UBEs.
+  uint8_t ube_protection;
+
+  // Jitter entropy state. NULL if not the per-process seed DRBG.
+  struct rand_data *jitter_ec;
```

Comment:
Does this make is_global redundant? Is that the same as `jitter_ec != null`? I'm thinking if this is a [don't repeat yourself](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and which of these two variables is authoritative?